hawkular-bus-common/src/main/java/org/hawkular/bus/common/BasicMessage.java

```diff
@@ -84,11 +89,49 @@ public void setMessageId(MessageId messageId) {
      * @return the message ID of the correlated message
      */
     public MessageId getCorrelationId() {
-        return correlationId;
+        return _correlationId;
     }
 
     public void setCorrelationId(MessageId correlationId) {
-        this.correlationId = correlationId;
+        this._correlationId = correlationId;
+    }
+
+    /**
+     * The headers that were shipped along side of the message when the message was received.
+     * The returned map is an unmodifiable read-only view of the properties.
+     * Can return <code>null</code>.
+     *
+     * @return a read-only view of the name/value properties that came with the message as separate headers.
+     */
+    public Map<String, String> getHeaders() {
+        if (_headers == null) {
+            return null;
```

Comment:
Would it make sense to return  `Collections.emptyMap()` instead so that the client does not have to do the null check?